src/Contexts/VersionContext.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import axios from 'axios';
+import logger from '../Utilities/Logger';
+import { version as packageVersion } from '../../package.json';
+import localPreferences from '../Utilities/LocalPreferences';
+import moment from 'moment';
+import _ from 'lodash';
+
+const { versionCheck } = localPreferences;
+
+interface VersionInformation {
+    serverFrontend?: string;
+    currentFrontend: string;
+}
+
+const getVersionFile = async () => {
+    return axios.get<string>(`/version.txt?cache_bust=${new Date().getTime()}`);
+};
+
+
+const versionRefresh = async () => {
+    // To test in development make a version.txt in the public folder with a value that matches or does not match package.json (i.e. "1.5.5")
+    // Then comment out this check
+    // version.txt in production is generated in post build
+    if (process.env.NODE_ENV !== 'production') {
+        logger.debug('Version Check: No cache buster in development');
+        return;
+    }
+    try {
+        const versionAxiosResponse = await getVersionFile();
+        const versionResponse = versionAxiosResponse.data;
+        if (packageVersion.trim() !== versionResponse.trim()) {
+            logger.warn(`Version Check: Version mismatch packageVersion ${packageVersion} !== versionResponse ${versionResponse}`);
+            const versionCheckDate = versionCheck.nextCheckDate;
+            let shouldReload = false;
+            if (_.isNil(versionCheckDate)) {
+                shouldReload = true;
+                logger.debug('Version Check: Time to try');
+            } else if (moment().isAfter(moment(versionCheckDate))) {
+                shouldReload = true;
+                logger.debug(`Version Check: Time to try again ${versionCheckDate}`);
+            } else {
+                shouldReload = false;
+                logger.error(`Version Check: Version mismatch packageVersion ${packageVersion} !== versionResponse ${versionResponse}. Too soon to try again ${versionCheckDate}`);
+            }
+    
+            if (shouldReload) {
+                // set the next time to the future
+                versionCheck.nextCheckDate = moment().utc().add(1, 'days').format();
+                // Using the reload flag is deprecated, however not using it did not seem to work, trying it out in case browsers support it
+                window.location.reload(true);
```

Comment:
A lot of this logic is to protect against infinite reloads, most of this logic should be deleted since we will not reload for the user at all

Reply:
https://github.com/rederlyhq/frontend/pull/523#discussion_r640613611